app/donate/page.tsx

```diff
@@ -0,0 +1,430 @@
+"use client"; 
+
+import React from 'react';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { z } from "zod"
+import { Button } from "@/components/ui/button"
+import { zodResolver } from "@hookform/resolvers/zod"
+import Image from 'next/image';
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form"
+import { Input } from "@/components/ui/input"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select"
+
+const Donate = () => {
+
+  const cookedFormSchema = z.object({
+    foodName: z.string().min(2, {
+      message: "Food name must be at least 2 characters.",
+    }),
+    timePrepared: z.string().nonempty({
+      message: "Date and Time of Preparation is required.",
+    }),
+    specialHandling: z.string(),
+    numberOfServings: z.number().min(1, {
+      message: "Number of servings must be at least 1.",
+    }),
+    consumeByTiming: z.string().nonempty({
+      message: "Consume By Timing is required."
+    }), 
+    foodImages: z.array(z.instanceof(File))
+  });
+
+  const nonCookedFormSchema = z.object({
+    foodName: z.string().min(2, {
+      message: "Food name must be at least 2 characters.",
+    }),
+    expiryDate: z.string().nonempty({
+      message: "Expiry Date is required.",
+    }),
+    foodCategory: z.string().nonempty({
+      message: "Food Category is required.",
+    }),
+    specialHandling: z.string(),
+    quantity: z.number().min(1, { 
+      message: "Quantity of food must be at least 1."
+    }),
+    foodImages: z.array(z.instanceof(File))
+  });
+
+  const cookedForm = useForm<z.infer<typeof cookedFormSchema>>({
+    resolver: zodResolver(cookedFormSchema),
+    defaultValues: {
+      foodName: "",
+      timePrepared: "",
+      consumeByTiming: "",
+      specialHandling: "",
+      numberOfServings: 0,
+      foodImages: []
+    },
+  });
+
+  const nonCookedForm = useForm<z.infer<typeof nonCookedFormSchema>>({
+    resolver: zodResolver(nonCookedFormSchema),
+    defaultValues: {
+      foodName: "",
+      expiryDate: "",
+      foodCategory: "",
+      specialHandling: "",
+      quantity: 0,
+      foodImages: []
+    },
+  });
+
+  function onSubmit(values: any) {
+    console.log(values);
```

Comment:
Is this console log statement supposed to be here?

Reply:
Oops no need to be there! Just added in to check that my data is correct when testing the submit button, I'll remove this tomorrow after I push my new donation form fields too

Reply:
Ah okay, thank you!